intro+ testes (#1)
VitorBSP/Trabalho-Final_TS: relatorio.Rmd

# Visualização dos dados

> No princípio, iremos realizar uma análise visual da Série Temporal para buscar ideiais inicias sobre o comportamento da série, visto a importância de entender seu desenvolvimento ao longo do tempo. Assim a Figura \@ref(fig:f1) irá fornecer esse vislumbre:

Reviewer: ```suggestion
Primeiramente é necessário  realizar uma análise visual da Série Temporal para buscar ideiais iniciais sobre seu comportamento, visto a importância de entender seu desenvolvimento ao longo do tempo. Assim a Figura \@ref(fig:f1) irá fornecer esse vislumbre:
```

```{r }
df <- read_delim("serie_1.txt", " ", col_names = F) |> rename(y = X1)
dfts <- ts(df, frequency = 12, start = c(0, 1))
```

```{r f1, fig.cap="Dados simulados ao longo do tempo"}
autoplot(dfts) +
  theme_minimal() +
  labs(title = "Gráfico da Série Temporal",
       x = "Meses", y = "Valores")
```

> É perceptível a provável existência de *outliers*, dado que apenas 3 pontos estão fora do intervalo de -5 a 5, conforme indica a Figura \@ref(fig:f1), pois temos  assim devemos investigá-los cuidadosamente. Assim como, não é evidente a existência de sazonalidade, tendência determísnistica ou estocástica, por isso, investigaremos esses as aspectos nessa e na próxima seção. Para melhorar a noção sobre os possíveis outliers também podemos analisar as medidas de tendência central e de dispersão, dadas pela Tabela \@ref(tab:t1).

Reviewer: ```suggestion
É perceptível a provável existência de *outliers*, dado que apenas 3 pontos estão fora do intervalo de -5 a 5, conforme indica a Figura \@ref(fig:f1), assim devemos investigá-los cuidadosamente. Deste modo não é evidente a existência de sazonalidade, tendência determínistica ou estocástica, por isso, investigaremos esses  aspectos nessa e na seção seguinte. Para melhorar a concepção sobre os possíveis *outliers* também podemos analisar as medidas de tendência central e de dispersão, dadas pela Tabela \@ref(tab:t1).
```

```{r t1}
df |>
  mypdf1::pdf1_summary() |>
  mypdf1::pdf1_tbl("Medidas Resumo")
```

> Desse modo é notório a diferença entre média, mediana e desvio padrão com o mínimo e máximo, como descreve a Tabela \@ref(tab:t1) também pode-se perceber a diferença entre os quartils (com o primeiro quartil, `r quantile(dfts, 0.25)`, e o terceiro é `r quantile(dfts, 0.75)`), assim reforça a possibilidade de existência de *outliers* pois temos uma distância considerável entre o primeiro quartil e o mínimo, assim como entre o terceiro quartil e o máximo.

Reviewer: ```suggestion
Desse modo é notório a diferença entre média, mediana e desvio padrão com o mínimo e máximo, como descreve a Tabela \@ref(tab:t1), também pode-se perceber a diferença entre os quartils (com o primeiro quartil, `r quantile(dfts, 0.25)`, e o terceiro é `r quantile(dfts, 0.75)`), assim reforça a possibilidade de existência de *outliers* pois temos uma distância considerável entre o primeiro quartil e o mínimo, assim como entre o terceiro quartil e o máximo.
```

## Outliers
Nessa subseção iremos avaliar efetivamente a existência de *outliers*, com o gráfico de *boxplot* e o Teste de Rosner, assim iremos começar com o boxplot na Figura \@ref(fig:f2) para auxiliar o teste.  

```{r f2, fig.cap="Boxplot da Série Temporal"}
dfts |>
  ggplot(aes(y=y, fill = y)) +
  geom_boxplot(show.legend = F) +
  scale_fill_brewer(palette = "Set2") +
  theme_minimal()+
  labs(x = "Meses", y = "Valores")
```

> Fora da área da caixa na Figura \@ref(fig:f2), temos 5 pontos, o que de acordo com a estrutura do *boxplot*, são os outliers, assim a estrutura do gráfico encontrou 5 possíveis *outliers*, que iremos avaliar com o Teste de Rosner na Tabela \@ref(tab:t2).

Reviewer: ```suggestion
Fora da área da caixa na Figura \@ref(fig:f2), temos 5 pontos, o que de acordo com a estrutura do *boxplot*, são os *outliers*, assim a estrutura do gráfico encontrou 5 possíveis *outliers*, que iremos avaliar com o Teste de Rosner na Tabela \@ref(tab:t2).
```

dfts1 <- ts(df1, frequency = 12, start = c(0, 1))

dados<-ts(dados3)
```

```{r f3, fig.cap="Dados simulados ao longo do tempo com remoção de outliers"}
autoplot(dfts1) +
  theme_minimal() +
  labs(title = "Gráfico da Série Temporal",
       x = "Meses", y = "Valores")
```
Analisando a Figura \@ref(fig:f3), é perceptível sua diferença em relação a Figura \@ref(fig:f1), nota-se principalmente que não existe mais o achatamento visual do gráfico, bem como o desaparecimento de picos e declives, devido a remoção dos outliers.

## Decomposição da Série Temporal

Para a finalização da visualização das particularidades da série temporal, utilizaremos a sua decomposição. Iniciando pela decisão de qual decomposição será utilizada: aditiva ou multiplicativa. A decisão será tomada com a soma do módulo dos resíduos da função de autocorrelação da decomposição, escolhendo aquele que possui a menor soma.
```{r}
res1 <- decompose(dfts1, type = "additive") |> with(random) |> tseries::na.remove()
res2 <- decompose(dfts1, type = "multiplicative") |> with(random) |> tseries::na.remove()

s1 <- sum(abs(acf(res1, plot = FALSE)$acf))
s2 <- sum(abs(acf(res2, plot = FALSE)$acf))
```

> A soma modular do resíduo aditivo do FAC é `r s1` e a do resíduo multiplicativo `r s2`, assim a soma do resíduo multiplicativo é menor, por isso utilizaremos ele no trabalho, assim na Figura \@ref(fig:f4) teremos a decomposição da Série Temporal.

Reviewer: Não acho *soma modular* uma expressão adequada tbh

Os gráficos de seção anterior evidenciaram a possibilidade da série não ser estacionária, assim faz-se necessário verificar se  vale de fato para o processo estocástico, com isso  vamos precisar aplicar testes de hipóteses para averiguar  algumas propriedades, como existência de tendência e  sazonalidade.

## Tendência 

Tendência refere-se a um algum  comportamento   não - estocástico da série em algum momento do tempo, se tal comportamento só acontece em alguns momentos especifícos do tempo, chamamos de tendência estocástica, do contrário é dita determinística. Vamos começar pelos testes de tendência determinística, que em termos sumarizados possuem como hipótese:

```{r}
kendal <- trend::mk.test(dfts1) |> with(p.value)
wald <- trend::ww.test(dfts1) |> with(p.value)
stuart <- trend::cs.test(dfts1) |> with(p.value)
tab1 <- data.frame("Testes" =  c("Cox-Stuart", "Wald-Wolfowitz", "Mann-Kendall"),
                   "P-valor" = c("<0.001", "<0.001", "<0.001"))
```

```{r t3}
tab1 %>%
  mypdf1::pdf1_tbl("Testes de Tendência Determinística")
```

> Com os teste de tendência determinística, com nível de significância de 5%, obtemos p-valores menores que 0.001, conforme mostrado pela Tabela \@ref(tab:t3), assim há evidências para rejeitar a hipótese $H_0$, então concluímos que existe tendência determinística na série temporal. 

Reviewer: *com nível de significância de 5%, obtemos p-valores menores que 0.001*, ok pra pra que falar 5% se está comparando com 0.001??

## Sazonalidade
Sazonalidade acontece quando a série possui um comportamento que se repete frequencialmente, vamos nessa subseção testar se existe sazonalidade nas avaliação dos episódios, assim temos como hipótese:
$$
H_0:\text{A série não possui sazonalidade}
$$
Os testes aplicados  foram Kruskal-Wallis, Friedman e Autocorrelação em lags Sazonais, assim gerando a seguite tabela:

```{r}
krusk <- seastests::kw(dfts1, freq = 12) |> with(Pval)
friedman <- seastests::fried(dfts1, freq = 12) |> with(Pval)
tqs <- seastests::qs(dfts1, freq = 12) |> with(Pval)
tab2 <- data.frame("Testes" = c("Kruskal-Wallis", "Friedman", "Autocorrelação em lags Sazonais"),
                   "P-valor" = c(krusk, friedman, tqs))
```

```{r t4}
tab2 %>%
  mypdf1::pdf1_tbl("Testes de Sazonalidade")
```

> Com os teste de sazonalidade, com nível de significância de 5%, obtemos p-valores maiores que 0.05, conforme mostrado pela Tabela \@ref(tab:t4), assim não há evidências para rejeitar a hipótese $H_0$, então concluímos que não há sazonalidade na série temporal. 

Reviewer: Mesma coisa aqui, refere-se *com nível de significância de 5%*, ai depois fala que o p-valor foi maior que ..
